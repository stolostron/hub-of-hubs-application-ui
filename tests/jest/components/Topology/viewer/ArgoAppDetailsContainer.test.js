// Copyright (c) 2021 Red Hat, Inc.
// Copyright Contributors to the Open Cluster Management project
"use strict";

import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import ArgoAppDetailsContainer from "../../../../../src-web/components/Topology/viewer/ArgoAppDetailsContainer";

const locale = "US-en";
window.open = () => {}; // provide an empty implementation for window.open

describe("ArgoAppDetailsContainer with no apps", () => {
  const mockData = {
    argoAppList: [],
    argoAppDetailsContainerControl: {
      argoAppDetailsContainerData: {
        page: 1,
        startIdx: 0,
        argoAppSearchToggle: false,
        expandSectionToggleMap: new Set(),
        selected: undefined,
        selectedArgpAppList: [],
        isLoading: false
      },
      handleArgoAppDetailsContainerUpdate: jest.fn(),
      handleErrorMsg: jest.fn()
    }
  };
  it("renders as expected", () => {
    const component = renderer.create(
      <ArgoAppDetailsContainer
        argoAppList={mockData.argoAppList}
        locale={locale}
        argoAppDetailsContainerControl={mockData.argoAppDetailsContainerControl}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("ArgoAppDetailsContainer with no apps", () => {
  const mockData = {
    argoAppList: [
      {
        name: "test1",
        cluster: "local-cluster",
        namespace: "ns1",
        destinationCluster: "cluster1",
        detinationNamespace: "test1-ns"
      },
      {
        name: "test2",
        cluster: "local-cluster",
        namespace: "ns2",
        destinationCluster: "cluster2",
        detinationNamespace: "test2-ns"
      }
    ],
    argoAppDetailsContainerControl: {
      argoAppDetailsContainerData: {
        page: 1,
        startIdx: 0,
        argoAppSearchToggle: false,
        expandSectionToggleMap: new Set(),
        selected: undefined,
        selectedArgpAppList: [],
        isLoading: false
      },
      handleArgoAppDetailsContainerUpdate: jest.fn(),
      handleErrorMsg: jest.fn()
    }
  };
  it("renders as expected", () => {
    const component = renderer.create(
      <ArgoAppDetailsContainer
        argoAppList={mockData.argoAppList}
        locale={locale}
        argoAppDetailsContainerControl={mockData.argoAppDetailsContainerControl}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("ArgoAppDetailsContainer test functions", () => {
  const mockData = {
    argoAppList: [
      {
        name: "test1",
        cluster: "local-cluster",
        namespace: "ns1",
        destinationCluster: "cluster1",
        detinationNamespace: "test1-ns"
      },
      {
        name: "test2",
        cluster: "local-cluster",
        namespace: "ns2",
        destinationCluster: "cluster2",
        detinationNamespace: "test2-ns"
      }
    ],
    argoAppDetailsContainerControl: {
      argoAppDetailsContainerData: {
        page: 1,
        startIdx: 0,
        argoAppSearchToggle: false,
        expandSectionToggleMap: new Set(),
        selected: undefined,
        selectedArgpAppList: [],
        isLoading: false
      },
      handleArgoAppDetailsContainerUpdate: jest.fn(),
      handleErrorMsg: jest.fn()
    }
  };

  let wrapper;
  beforeEach(
    () =>
      (wrapper = shallow(
        <ArgoAppDetailsContainer
          argoAppList={mockData.argoAppList}
          locale={locale}
          argoAppDetailsContainerControl={
            mockData.argoAppDetailsContainerControl
          }
        />
      ))
  );

  it("works as expected", () => {
    const instance = wrapper.instance();
    instance.toggleLinkLoading();
    instance.handleExpandSectionToggle(0);
    instance.handleSelection({}, "test1");
    instance.handleSelection({}, undefined);
    instance.handleSelectToggle();
    instance.handleSelectionClear();
    instance.handleFirstClick();
    instance.handleLastClick();
    instance.handleNextClick({}, 1);
    instance.handlePreviousClick({}, 1);
    instance.handlePageInput({}, 1);
    instance.handleKeyPress(
      {
        action: "open_link",
        targetLink: "https://test"
      },
      {
        key: "Enter"
      }
    );
    instance.handleKeyPress(
      {
        action: "open_link",
        targetLink: "https://test"
      },
      {
        key: "Any"
      }
    );
  });
});
