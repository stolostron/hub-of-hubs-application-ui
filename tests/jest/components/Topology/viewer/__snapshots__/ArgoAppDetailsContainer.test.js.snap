// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArgoAppDetailsContainer with no apps renders as expected 1`] = `
<div
  className="appDetails"
>
  <div
    className="pf-c-select"
    data-ouia-component-id="OUIA-Generated-Select-typeahead-1"
    data-ouia-component-type="PF4/Select"
    data-ouia-safe={true}
  >
    <div
      className="pf-c-select__toggle pf-m-typeahead"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <div
        className="pf-c-select__toggle-wrapper"
      >
        <input
          aria-activedescendant={null}
          aria-label="Find application"
          autoComplete="off"
          className="pf-c-form-control pf-c-select__toggle-typeahead"
          disabled={false}
          id="pf-select-toggle-id-0-select-typeahead"
          onChange={[Function]}
          onClick={[Function]}
          placeholder="Find application"
          type="text"
          value=""
        />
      </div>
      <button
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-label="Options menu"
        aria-labelledby="app-select-id-1 pf-select-toggle-id-0"
        className="pf-c-button pf-c-select__toggle-button pf-m-plain"
        disabled={false}
        id="pf-select-toggle-id-0"
        onClick={[Function]}
        tabIndex={-1}
        type="button"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          className="pf-c-select__toggle-arrow"
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="spacer"
  />
  <div
    className="spacer"
  />
  <dl
    aria-label=""
    className="pf-c-accordion"
  />
</div>
`;

exports[`ArgoAppDetailsContainer with no apps renders as expected 2`] = `
<div
  className="appDetails"
>
  <div
    className="pf-c-select"
    data-ouia-component-id="OUIA-Generated-Select-typeahead-2"
    data-ouia-component-type="PF4/Select"
    data-ouia-safe={true}
  >
    <div
      className="pf-c-select__toggle pf-m-typeahead"
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <div
        className="pf-c-select__toggle-wrapper"
      >
        <input
          aria-activedescendant={null}
          aria-label="Find application"
          autoComplete="off"
          className="pf-c-form-control pf-c-select__toggle-typeahead"
          disabled={false}
          id="pf-select-toggle-id-1-select-typeahead"
          onChange={[Function]}
          onClick={[Function]}
          placeholder="Find application"
          type="text"
          value=""
        />
      </div>
      <button
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-label="Options menu"
        aria-labelledby="app-select-id-1 pf-select-toggle-id-1"
        className="pf-c-button pf-c-select__toggle-button pf-m-plain"
        disabled={false}
        id="pf-select-toggle-id-1"
        onClick={[Function]}
        tabIndex={-1}
        type="button"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          className="pf-c-select__toggle-arrow"
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="spacer"
  />
  <div
    className="spacer"
  />
  <dl
    aria-label=""
    className="pf-c-accordion"
  >
    <div
      className="appDetailItem"
      style={
        Object {
          "borderBottom": "1px solid #D2D2D2",
          "borderTop": "1px solid #D2D2D2",
        }
      }
    >
      <dt>
        <button
          aria-expanded={false}
          className="pf-c-accordion__toggle"
          id="test1"
          onClick={[Function]}
          type="button"
        >
          <span
            className="pf-c-accordion__toggle-text"
          >
            <svg
              fill="#878D96"
              height="12px"
              width="12px"
            >
              <use
                className="label-icon"
                href="#diagramIcons_pending"
              />
            </svg>
            <span
              style={
                Object {
                  "paddingRight": "10px",
                }
              }
            />
            test1
          </span>
          <span
            className="pf-c-accordion__toggle-icon"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </span>
        </button>
      </dt>
      <dd
        aria-label=""
        className="pf-c-accordion__expanded-content"
        hidden={true}
        id=""
      >
        <div
          className="pf-c-accordion__expanded-content-body"
        >
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            <span
              className="link sectionLabel"
              id="linkForNodeAction"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="0"
            >
              Launch Argo editor
              <svg
                height="12px"
                style={
                  Object {
                    "marginLeft": "8px",
                    "stroke": "#0066CC",
                  }
                }
                width="12px"
              >
                <use
                  className="label-icon"
                  href="#diagramIcons_carbonLaunch"
                />
              </svg>
            </span>
          </span>
          <div
            className="spacer"
          />
          <span
            className="label sectionLabel"
            style={
              Object {
                "fontSize": "0.75rem",
                "paddingLeft": "1rem",
              }
            }
          >
            Details
          </span>
          <div
            className="spacer"
          />
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="link sectionLabel"
              id="linkForNodeAction"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="0"
            >
              View resource YAML
              <svg
                height="12px"
                style={
                  Object {
                    "marginLeft": "8px",
                    "stroke": "#0066CC",
                  }
                }
                width="12px"
              >
                <use
                  className="label-icon"
                  href="#diagramIcons_open-new-tab"
                />
              </svg>
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Created on
              :
               
            </span>
            <span
              className="value"
            >
              local-cluster
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Destination cluster
              :
               
            </span>
            <span
              className="value"
            >
              cluster1
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Destination namespace
              :
               
            </span>
            <span
              className="value"
            />
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Status
              :
               
            </span>
            <span
              className="value"
            />
          </div>
          <div
            className="spacer"
          />
        </div>
      </dd>
      <span
        style={
          Object {
            "display": "block",
          }
        }
      >
        <span
          className="link sectionLabel"
          id="linkForNodeAction"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          Launch Argo editor
          <svg
            height="12px"
            style={
              Object {
                "marginLeft": "8px",
                "stroke": "#0066CC",
              }
            }
            width="12px"
          >
            <use
              className="label-icon"
              href="#diagramIcons_carbonLaunch"
            />
          </svg>
        </span>
      </span>
    </div>
    <div
      className="appDetailItem"
      style={
        Object {
          "borderBottom": "1px solid #D2D2D2",
        }
      }
    >
      <dt>
        <button
          aria-expanded={false}
          className="pf-c-accordion__toggle"
          id="test2"
          onClick={[Function]}
          type="button"
        >
          <span
            className="pf-c-accordion__toggle-text"
          >
            <svg
              fill="#878D96"
              height="12px"
              width="12px"
            >
              <use
                className="label-icon"
                href="#diagramIcons_pending"
              />
            </svg>
            <span
              style={
                Object {
                  "paddingRight": "10px",
                }
              }
            />
            test2
          </span>
          <span
            className="pf-c-accordion__toggle-icon"
          >
            <svg
              aria-hidden={true}
              aria-labelledby={null}
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 256 512"
              width="1em"
            >
              <path
                d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
              />
            </svg>
          </span>
        </button>
      </dt>
      <dd
        aria-label=""
        className="pf-c-accordion__expanded-content"
        hidden={true}
        id=""
      >
        <div
          className="pf-c-accordion__expanded-content-body"
        >
          <span
            style={
              Object {
                "display": "block",
              }
            }
          >
            <span
              className="link sectionLabel"
              id="linkForNodeAction"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="0"
            >
              Launch Argo editor
              <svg
                height="12px"
                style={
                  Object {
                    "marginLeft": "8px",
                    "stroke": "#0066CC",
                  }
                }
                width="12px"
              >
                <use
                  className="label-icon"
                  href="#diagramIcons_carbonLaunch"
                />
              </svg>
            </span>
          </span>
          <div
            className="spacer"
          />
          <span
            className="label sectionLabel"
            style={
              Object {
                "fontSize": "0.75rem",
                "paddingLeft": "1rem",
              }
            }
          >
            Details
          </span>
          <div
            className="spacer"
          />
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="link sectionLabel"
              id="linkForNodeAction"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex="0"
            >
              View resource YAML
              <svg
                height="12px"
                style={
                  Object {
                    "marginLeft": "8px",
                    "stroke": "#0066CC",
                  }
                }
                width="12px"
              >
                <use
                  className="label-icon"
                  href="#diagramIcons_open-new-tab"
                />
              </svg>
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Created on
              :
               
            </span>
            <span
              className="value"
            >
              local-cluster
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Destination cluster
              :
               
            </span>
            <span
              className="value"
            >
              cluster2
            </span>
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Destination namespace
              :
               
            </span>
            <span
              className="value"
            />
          </div>
          <div
            className="sectionContent borderLeft"
          >
            <span
              className="label sectionLabel"
            >
              Status
              :
               
            </span>
            <span
              className="value"
            />
          </div>
          <div
            className="spacer"
          />
        </div>
      </dd>
      <span
        style={
          Object {
            "display": "block",
          }
        }
      >
        <span
          className="link sectionLabel"
          id="linkForNodeAction"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        >
          Launch Argo editor
          <svg
            height="12px"
            style={
              Object {
                "marginLeft": "8px",
                "stroke": "#0066CC",
              }
            }
            width="12px"
          >
            <use
              className="label-icon"
              href="#diagramIcons_carbonLaunch"
            />
          </svg>
        </span>
      </span>
    </div>
  </dl>
</div>
`;
